api/odl2.py

```diff
@@ -1,80 +1,78 @@
 from __future__ import annotations
 
 import logging
-from typing import TYPE_CHECKING
+from typing import TYPE_CHECKING, List, Optional
 
-from contextlib2 import contextmanager
 from flask_babel import lazy_gettext as _
 from webpub_manifest_parser.odl import ODLFeedParserFactory
 from webpub_manifest_parser.opds2.registry import OPDS2LinkRelationsRegistry
 
 from api.circulation_exceptions import PatronHoldLimitReached, PatronLoanLimitReached
-from api.odl import ODLAPI, ODLImporter
+from api.odl import ODLAPI, ODLImporter, ODLSettings
+from core.integration.settings import (
+    ConfigurationFormItem,
+    ConfigurationFormItemType,
+    FormField,
+)
 from core.metadata_layer import FormatData
 from core.model import Edition, RightsStatus
-from core.model.configuration import (
-    ConfigurationAttributeType,
-    ConfigurationFactory,
-    ConfigurationMetadata,
-    ConfigurationStorage,
-    ExternalIntegration,
-    HasExternalIntegration,
-)
+from core.model.configuration import ExternalIntegration, HasExternalIntegration
 from core.opds2_import import OPDS2Importer, OPDS2ImportMonitor, RWPMManifestParser
-from core.opds_import import OPDSImporterConfiguration
 from core.util import first_or_default
 from core.util.datetime_helpers import to_utc
 
 if TYPE_CHECKING:
     from core.model.patron import Patron
 
 
-class ODL2APIConfiguration(OPDSImporterConfiguration):
-    skipped_license_formats = ConfigurationMetadata(
-        key="odl2_skipped_license_formats",
-        label=_("Skipped license formats"),
-        description=_(
-            "List of license formats that will NOT be imported into Circulation Manager."
-        ),
-        type=ConfigurationAttributeType.LIST,
-        required=False,
+class ODL2Settings(ODLSettings):
+    odl2_skipped_license_formats: Optional[List[str]] = FormField(
```

Comment:
All the odl2 settings need to maintain the configuation key names, hence the `odl2` prefix